lesson28/src/js/changepassword.js

```diff
@@ -0,0 +1,45 @@
+import { checkFormValidityInBlur } from "./modules/validation";
+
+const urlParameter = Object.fromEntries(new URLSearchParams(window.location.search));
+const currentPageToken = urlParameter.token;
+const registeredToken = localStorage.getItem("passwordReissueToken");
+
+if(currentPageToken !== registeredToken) window.location.href = "./../notautherize.html";
+
+
+const passwordOfInput = document.querySelector(".js-form-password");
+const confirmPasswordOfInput = document.querySelector(".js-form-confirm-password");
+const formElements = [passwordOfInput, confirmPasswordOfInput];
+const submitButton = document.querySelector(".js-submit-button");
+const eyeIcons = document.querySelectorAll(".js-eye-icon");
+
+formElements.forEach(element => {
+    element.classList.add("invalid");
+
+    element.addEventListener("blur", (e) => {
+        checkFormValidityInBlur(submitButton, e.target);
+    });
+});
+
+eyeIcons.forEach(icon => {
+    icon.addEventListener("click", (e) => togglePasswordDisplay(e.target))
+})
+
+const togglePasswordDisplay = target => {
+    const selectedInput = target.nextElementSibling;
+    if (selectedInput.type === "password"){
+        selectedInput.type = "text";
+        target.classList.add("is-open");
+    } else {
+        selectedInput.type = "password";
+        target.classList.remove("is-open");
+    }
```

Comment:
IMO
You can also write like this
```suggestion
  const selectedInput = target.nextElementSibling;
  selectedInput.type = selectedInput.type === "password" ? "text" : "password";
  target.classList.toggle("is-open");```

Reply:
Thank you for always suggesting smart code!
I hadn't thought of it.
So with the ternary operator and the toggle method I could have written it in 3 lines!
I learned a lot.
[0cb80dd](https://github.com/haru-programming/JavaScript-lessons/pull/35/commits/0cb80dd3541e504f4f971e253aa0b1895cf35d42)
[CodeSandBox](https://codesandbox.io/s/github/haru-programming/JavaScript-lessons/tree/feature/lesson28-3/lesson28)